pkg/issuer/acme/setup.go

```diff
@@ -122,7 +148,7 @@ func (a *Acme) Setup(ctx context.Context) error {
 	//  and remove them when the corresponding issuer is updated/deleted.
 	a.accountRegistry.RemoveClient(string(a.issuer.GetUID()))
```

Comment:
The only reason I can imagine as to why this is called here, rather than at the start, could be so that in a scenario where there is already a valid client in the cache, but in `Setup` we somehow fail to retrieve the secret with the account key, the orders and challenges controllers can keep using the valid client in the cache.
I am not sure whether this is how we intend for this to work.

I could either move it up or add a 'TODO' type of comment to investigate whether we want to move it up.

Reply:
But if the Secret is not found, then the Issuer resource should be marked not-ready and the ACME certificate request controller should see that the Issuer is not-ready and not proceed.

A TODO comment would be a helpful reminder that this is not a well understood piece of code.